packages/nvidia_nat_test/src/nat/test/utils.py

```
@@ -89,6 +91,71 @@ async def run_workflow(*,
+                         port: int = 8000,
+                         pipeline_timeout: int = 60,
+                         request_timeout: int = 30) -> dict:
+    """
+    Execute a workflow using `nat serve`, and issue a POST request to the `/generate` endpoint with the given question.
+
+    Intendted to be analogous to `run_workflow` but for the REST API serving mode.
+    """
+    import requests
+    workflow_url = f"http://localhost:{port}"
+    workflow_cmd = ["nat", "serve", "--port", str(port), "--config_file", str(config_path.absolute())]
+    proc = subprocess.Popen(workflow_cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
```

Ruff (0.14.8)

`subprocess` call: check for execution of untrusted input

(S603)